contracts/BridgeMessage.sol

```diff
@@ -0,0 +1,508 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity ^0.8.15;
+
+// ============ External Imports ============
+import {TypedMemView} from "@summa-tx/memview-sol/contracts/TypedMemView.sol";
+
+library BridgeMessage {
+    // ============ Libraries ============
+
+    using TypedMemView for bytes;
+    using TypedMemView for bytes29;
+
+    // ============ Enums ============
+
+    // WARNING: do NOT re-write the numbers / order
+    // of message types in an upgrade;
+    // will cause in-flight messages to be mis-interpreted
+    // The Types enum has to do with the TypedMemView library and it defines
+    // the types of `views` that we use in BridgeMessage. A view is not interesting data
+    // itself, but rather it points to a specific part of the memory where
+    // the data we care about live. When we give a `type` to a view, we define what type
+    // is the data it points to, so that we can do easy runtime assertions without
+    // having to fetch the whole data from memory and check for ourselves. In BridgeMessage.sol
+    // the types of `data` we can have are defined in this enum and may belong to different taxonomies.
+    // For example, a `Message` includes a `TokenId` and an Action, either a `Transfer` or a `TransferToHook`.
+    // The Message is a different TYPE of data than a TokenId or Transfer, as TokenId and Transfer live inside
+    // the message. For that reason, we define them as different data types and we add them to the same enum
+    // for ease of use.
+    enum Types {
+        Invalid, // 0
+        TokenId, // 1
+        Message, // 2
+        Transfer, // 3
+        DeprecatedFastTransfer, // 4
+        TransferToHook, // 5
+        ExtraData, // 6
+        UpdateSupply //7
+    }
+
+    // ============ Structs ============
+
+    // Tokens are identified by a TokenId:
+    // domain - 4 byte chain ID of the chain from which the token originates
+    // id - 32 byte identifier of the token address on the origin chain, in that chain's address format
+    struct TokenId {
+        uint32 domain;
+        bytes32 id;
+    }
+
+    // ============ Constants ============
+
+    uint256 private constant TOKEN_ID_LEN = 36; // 4 bytes domain + 32 bytes id
+    uint256 private constant IDENTIFIER_LEN = 1;
+    uint256 private constant TRANSFER_LEN = 97; // 1 byte identifier + 32 bytes recipient + 32 bytes amount + 32 bytes detailsHash
+    uint256 private constant MIN_TRANSFER_HOOK_LEN = 129; // 1 byte identifier + 32 bytes hook address + 32 bytes amount + 32 bytes detailsHash + 32 bytes sender + X bytes extraData
+
```

Comment:
This is missing the length for `updateSupply`

We'd need a `uint256 private constant UPDATE_SUPPLY_LEN = 33`

(1 byte identifier + 32 bytes totalSupply)